File: pset1/cash/cash.c
#include<stdio.h>
#include<cs50.h>
#include<math.h>

int main(void)
//Print the least number of coins for change owed the user
{
    float co;

    //Prompt user for change owed
    do
    {
        co = get_float("change owed: ");
    }
    while (co <= 0);

    int cents = round(co * 100);
    int output = 0;

    int qtrs = 0;
    int dmes = 0;
    int nckls = 0;
    int pnnies = 0;

    //Check to see if we can use quarters
    while (cents >= 25)
    {
        cents -= 25;
        output += 1;
        qtrs++;
    }

    //Check to see if we can use dimes
    while (cents < 25 && cents > 9)
    {
        cents -= 10;
        output += 1;
        dmes++;
    }

    //Check to see if we can use nickels
    while (cents < 10 && cents > 4)
    {
        cents -= 5;
        output += 1;
        nckls++;
    }

    //Check to see if we can use pennies
    while (cents < 5 && cents > 0)
    {
        cents -= 1;
        output += 1;
        pnnies++;
    }
    printf("%i\n", output); //Print the least number of coins
    printf("quarters: %i\ndimes: %i\nnickels: %i\npennies: %i\n", qtrs, dmes, nckls, pnnies);
}

File: pset6/cash/cash.py
from cs50 import get_float

while True:
    #Get user input for change owed
    change = get_float("change owed: ")
    if change > 0:
        break

cents = round(change * 100)
coins = 0

#Iterates through the change owed, always looking for the largest coin that will fit
while cents > 0:
    if cents > 24:
        coins += cents // 25
        cents = cents % 25
    elif cents < 25 and cents > 9:
        coins += cents // 10
        cents = cents % 10
    elif cents < 10 and cents > 4:
        coins += cents // 5
        cents = cents % 5
    elif cents < 5 and cents > 0:
        coins += cents // 1
        cents = cents % 1

print(coins)

#Alternative implementation

#while cents > 0:
#    if cents >= 25:
#        cents -= 25
#        coins += 1
#    elif cents < 25 and cents >= 10:
#        cents -= 10
#        coins += 1
#    elif cents < 10 and cents >= 5:
#        cents -= 5
#        coins += 1
#    elif cents < 5 and cents > 0:
#        cents -= 1
#        coins += 1

#print(coins)

